Cue/app/actions/library.js

```diff
@@ -17,4 +19,80 @@ async function loadLibrary(): PromiseAction {
   };
 }
 
-module.exports = { loadLibrary };
+function deleteDeck(uuid: string) : Action {
+  return {
+    type: 'DECK_DELETED',
+    uuid
+  }
+}
+
+function createDeck(name: string) : Action {
+  let deck = {
+    name,
+    uuid: uuidV4(),
+    public: false,
+    cards: [],
+    owner: CueApi.userId,
+    accession: 'private',
+  }
+  return {
+    type: 'DECK_CREATED',
+    deck,
+  }
+}
+
+function editDeck(deck: DeckChanges) : Action {
+  return {
+    type: 'DECK_EDITED',
+    deck
+  }
+}
+
+//attempts to sync each deck once, returns list of failed syncs
+function syncLibrary(localChanges): ThunkAction {
+  let failedSyncs = []
+  return (dispatch, getState) => {
+    let promises = []
+    localChanges.forEach(change =>{
+      promises.push(dispatch(syncDeck(change))
+        .catch(e => {
+          console.info('sync failed : ', change)
+          failedSyncs.push(change)
+        }))
+    })
+    Promise.all(promises).then(() => {
+      if (failedSyncs.length == 0) dispatch(loadLibrary())
+    })
+    console.log('failed syncs:  ', failedSyncs)
+    return failedSyncs
+  };
+}
+
+// deck should only contain changes in the deck to avoid sending excess data
```

Comment:
The `library` reducer doesn't do this diffing for edited cards right now, right?